package-res/resources/web/pentaho/config/util.js

```diff
@@ -0,0 +1,222 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../util/object",
+  "../lang/OperationInvalidError"
+], function(O, OperationInvalidError) {
+
+  "use strict";
+
+  /**
+   * Map of merge operation name to operation handler function.
+   *
+   * @type {Object.<string, function>}
+   * @see mergeSpecsOne
+   */
+  var _mergeHandlers = {
+    "replace": mergeSpecsOperReplace,
+    "merge": mergeSpecsOperMerge,
+    "add": mergeSpecsOperAdd
+  };
+
+  /**
+   * @classDesc A singleton class that contains utilities related to configuration.
+   *
+   * @name pentaho.config.Util
+   * @class
```

Comment:
This `mergeSpecs` utility method was in the local `pentaho.type.Util` package. I ended creating another local utilities class for the config package: `pentaho.config.Util`. I pondered whether this could be a method of one of the existing utility "pentaho.util.*" but couldn't figure out a good receiver. I pondered creating a new "pentaho/util/spec" utility module, but this would put into question the current rule that "spec is a sub-package of auxiliary, JSON-like interfaces"... Then, this package is also public, and `pentaho.util.*` is almost totally private, the exception being the `MessageBundle` package. There is also no precedent of one of the utility packages in `pentaho.util` having only one public method... Big mess.